### PR TITLE
Mistral prefill 128-4k

### PR DESCRIPTION
**Pull request by sraizada-tt**
### Ticket
#5337 

### Problem description
Support fast prefill for mistral7B model up to 4k sequence length.

### What's changed
Add prefill modules, tests, and demo integration for mistral7B. 

Missing prefill perf test. 

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10012202867)
- [x] [Nightly fast dispatch](https://github.com/tenstorrent/tt-metal/actions/runs/10076080318) (nightly GS failing, unrelated to this PR)
- [x] [Model perf](https://github.com/tenstorrent/tt-metal/actions/runs/10076088644)
- [x] [Model Demos](https://github.com/tenstorrent/tt-metal/actions/runs/10076785820/job/27860427558) (Mamba failing on n150, unrelated to this PR)

### CODE REVIEW

File: models/demos/wormhole/mistral7b/tt/mistral_rms_norm.py
Comment (yieldthought) [inline models/demos/wormhole/mistral7b/tt/mistral_rms_norm.py:1]: I think this file is unused now and should be removed

**Review by yieldthought — APPROVED** [review]

File: models/demos/wormhole/mistral7b/demo/demo.py
Comment (yieldthought) [inline models/demos/wormhole/mistral7b/demo/demo.py:218]: Was this intended?

**Review by yieldthought — APPROVED** [review]

File: models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py
Comment (tt-rkim) [inline models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py:315]: Please use this https://github.com/tenstorrent/tt-metal/pull/10371/files

File: tests/scripts/single_card/run_demos_single_card_wh_common.sh
Comment (tt-rkim) [inline tests/scripts/single_card/run_demos_single_card_wh_common.sh:15]: Could we take the opportunity to simplify this command and extract the demo out into a separate test that we can call like `WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/demos/wormhole/mistral7b/demo/demo_with_prefill.py` ?

Comment (mtairum) [reply tests/scripts/single_card/run_demos_single_card_wh_common.sh:15]: Yes. I'm working on that now. This will come with the use of the CI fixture 👍

Comment (mtairum) [reply tests/scripts/single_card/run_demos_single_card_wh_common.sh]: Just pushed changes addressing this for all Mistral tests and demo code. We will look into Mixtral in a separate PR.

**Comment by mtairum** [comment]
Just pushed a major refactor to all Mistral tests and demo code.

Running all pipelines (updated the PR description as well)

- [ ] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10012202867)
- [ ] [Nightly fast dispatch tests](https://github.com/tenstorrent/tt-metal/actions/runs/10012205432)
- [ ] [Single card Demo tests](https://github.com/tenstorrent/tt-metal/actions/runs/10012204512)
- [ ] [Model perf](https://github.com/tenstorrent/tt-metal/actions/runs/10012207970) (we don't include prefill perf test, but we made changes to codebase so we should guarantee it passes)

**Comment by mtairum** [comment]
@sraizada-tt Seeing this issue locally on Mistral perf test:

```
        assert len(x.shape) == 3
>       assert x.shape[2] == hidden_size
E       AssertionError

models/demos/wormhole/mistral7b/tt/mistral_common.py:118: AssertionError
```

**Comment by mtairum** [comment]
I've removed the KV cache PCC checks from Mistral attention and attention_prefill tests.

There's something weird going on and we will be tracking that in issue #10648 

Re-running all pipelines and will merge after they all pass:
- [x] [Nightly fast dispatch](https://github.com/tenstorrent/tt-metal/actions/runs/10076080318)
- [x] [Model perf](https://github.com/tenstorrent/tt-metal/actions/runs/10076088644)
- [ ] [Model Demos] (https://github.com/tenstorrent/tt-metal/actions/runs/10076785820/job/27860427558)

**Review by mtairum — APPROVED** [review]

File: models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py
Comment (tt-rkim) [reply models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py:315]: This is still not using the fixture?

File: models/demos/wormhole/mistral7b/demo/demo_with_prefill.py
Comment (tt-rkim) [inline models/demos/wormhole/mistral7b/demo/demo_with_prefill.py:132]: We'll still need to open an issue to centralize these directories, but that isn't blocking this PR

File: models/demos/wormhole/mistral7b/tests/test_mistral_decoder_prefill.py
Comment (tt-rkim) [inline models/demos/wormhole/mistral7b/tests/test_mistral_decoder_prefill.py:105]: Should we print out which outputs failed? Not the values, but the indices

File: models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py
Comment (mtairum) [reply models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py:315]: This PR only touches Mistral.

This change was a small bug I noticed and did a quick patch inside this PR. 

We will open a new PR with the changes on Mixtral side after this one.

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**
